docs/user-guide/102-policy-syntax.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Policy Syntax"
+tag: "user-guide"
+---
+
+The policy contains the following fields:
+- `capture`: Contains some simple expressions
+to evaluate the current network configuration and store the result at 
+variables.
+- `desiredState`: Contains network state yaml with references to `capture` entries.
+
+__Example:__
+```yaml
+{% include_absolute 'examples/bridge-on-default-gw-dhcp/policy.md' %}
+```
+
+## Capture syntax
+
+The capture is a map of capture entries with an identifier as the key that can 
+be referenced with ```capture.[id]```
+
+The map value is a capture entry expression that evaluates the current 
+network state and stores the result.
+
+The following is a semi-formal definition of the capture entry expression:
+
+```
+<letter> ::= "A" | "B" | "C" | "D" | "E" | "F" | "G"
+       | "H" | "I" | "J" | "K" | "L" | "M" | "N"
+       | "O" | "P" | "Q" | "R" | "S" | "T" | "U"
+       | "V" | "W" | "X" | "Y" | "Z" | "a" | "b"
+       | "c" | "d" | "e" | "f" | "g" | "h" | "i"
+       | "j" | "k" | "l" | "m" | "n" | "o" | "p"
+       | "q" | "r" | "s" | "t" | "u" | "v" | "w"
+       | "x" | "y" | "z"
+<digit> ::= [0-9]
+<number> ::= <digit>+
+<identifier> ::= <letter> ( <digit> | "-" | <letter> )*
+<dot> ::= "."
+<path> ::= <identifier> ( <dot> <identity> )*
+<string> ::= "\"" (<all characters>)* "\""
+<bool> ::= "true" | "false"
+
+<captureid> ::= <identity>
+<capturepath> ::= "capture" <dot> <captureid> <path>
+<eqoperator> ::= "=="
+<eqexpression> ::= <path> <eqoperator> (<string> | <number> | <bool> | <capturepath>)
+<eqoperator> ::= ":="
+<replaceexpression> ::= <path> <replaceoperator> (<string> | <number> | <bool> | <capturepath>)
+<expression> ::= <eqfilterexpression> | <replaceexpression>
+<pipe> ::= "|"
+<pipedexpression> ::= ( <capturepath> | <expression> ) <pipe> <expression>
```

Comment:
Currently pipe in expression is [not supported](https://github.com/nmstate/nmpolicy/blob/73d8e32eeec366dc825ccff4ecd9e2a273cdeae8/nmpolicy/internal/resolver/resolver.go#L173).